File: Baekjoon/Greedy/1946.py
# https://www.acmicpc.net/problem/1946

import sys
input = sys.stdin.readline

for _ in range(int(input())):
    n = int(input())
    data = []
    for _ in range(n):
        apply, interview = map(int, input().split())
        data.append((apply, interview))

    data.sort(key = lambda x: x[1])
    min = data[0][0]
    count = 1
    
    for i in range(1, n):
        if data[i][0] < min:
            min = data[i][0]
            count += 1

    print(count)

File: Baekjoon/Greedy/1946.cpp
// https://www.acmicpc.net/problem/1946

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        vector<pair<int, int>> a(n);
        for (int i = 0; i < n; i++) {
            cin >> a[i].first >> a[i].second;
        }
        sort(a.begin(), a.end());
        int ans = 1;
        int min = a[0].second;
        for (int i = 1; i < n; i++) {
            if (a[i].second < min) {
                min = a[i].second;
                ans += 1;
            }
        }
        cout << ans << '\n';
    }

    return 0;
}